wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/JpaWorkerDao.java

```diff
@@ -153,4 +159,33 @@ public void onEvent(BeforeWorkspaceRemovedEvent event) {
             }
         }
     }
+
+    @Singleton
+    public static class RemoveSystemPermissionsBeforeUserRemovedEventSubscriber implements EventSubscriber<BeforeUserRemovedEvent> {
+        @Inject
+        private EventService eventService;
+        @Inject
+        private WorkerDao    dao;
+
+        @PostConstruct
+        public void subscribe() {
+            eventService.subscribe(this);
+        }
+
+        @PreDestroy
+        public void unsubscribe() {
+            eventService.unsubscribe(this);
+        }
+
+        @Override
+        public void onEvent(BeforeUserRemovedEvent event) {
+            try {
+                for (WorkerImpl worker : dao.getWorkersByUser(event.getUser().getId())) {
+                    dao.removeWorker(worker.getInstanceId(), worker.getUserId());
+                }
+            } catch (Exception x) {
+                LOG.error(format("Couldn't remove permissions before user '%s' is removed", event.getUser().getId()), x);
```

Comment:
I think it would be better to be more verbose here, i mean, if you add a type of permission which can't be removed - it becomes clearer from the message what the problem is, wdyt ?